doc/api-examples/c/list_nodes.c

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+#include <dbus/dbus.h>
```

Comment:
Since we are using `sd-bus` it would make sense to provide examples using it for the examples as well. Its also easier to use than the reference implementation, making it more suitable for samples, I think.